flink-connector-aws/flink-connector-sqs/src/main/java/org/apache/flink/connector/sqs/sink/SqsSinkBuilder.java

```diff
@@ -0,0 +1,159 @@
+
+/**
+ * Builder to construct {@link SqsSink}.
+ *
+ * <p>The following example shows the minimum setup to create a {@link SqsSink} that writes String
+ * values to a SQS named sqsUrl.
+ *
+ * <pre>{@code
+ * Properties sinkProperties = new Properties();
+ * sinkProperties.put(AWSConfigConstants.AWS_REGION, "eu-west-1");
+ *
+ * SqsSink<String> sqsSink =
+ *         SqsSink.<String>builder()
+ *                 .setSqsUrl("sqsUrl")
+ *                 .setSqsClientProperties(sinkProperties)
+ *                 .setSerializationSchema(new SimpleStringSchema())
+ *                 .build();
+ * }</pre>
+ *
+ * <p>If the following parameters are not set in this builder, the following defaults will be used:
+ *
+ * <ul>
+ *   <li>{@code maxBatchSize} will be 10
+ *   <li>{@code maxInFlightRequests} will be 50
+ *   <li>{@code maxBufferedRequests} will be 5000
+ *   <li>{@code maxBatchSizeInBytes} will be 256 KB i.e. {@code 256 * 1000}
+ *   <li>{@code maxTimeInBufferMs} will be 5000ms
+ *   <li>{@code maxRecordSizeInBytes} will be 256 KB i.e. {@code 256 * 1000}
+ *   <li>{@code failOnError} will be false
+ * </ul>
+ *
+ * @param <InputT> type of elements that should be persisted in the destination
+ */
+@PublicEvolving
+public class SqsSinkBuilder<InputT>
+        extends AsyncSinkBaseBuilder<InputT, SendMessageBatchRequestEntry, SqsSinkBuilder<InputT>> {
+
+    private static final int DEFAULT_MAX_BATCH_SIZE = 10;
+    private static final int DEFAULT_MAX_IN_FLIGHT_REQUESTS = 50;
+    private static final int DEFAULT_MAX_BUFFERED_REQUESTS = 5_000;
+    private static final long DEFAULT_MAX_BATCH_SIZE_IN_B = 256 * 1000;
+    private static final long DEFAULT_MAX_TIME_IN_BUFFER_MS = 5000;
+    private static final long DEFAULT_MAX_RECORD_SIZE_IN_B = 256 * 1000;
+    private static final boolean DEFAULT_FAIL_ON_ERROR = false;
+    private static final Protocol DEFAULT_HTTP_PROTOCOL = HTTP1_1;
+
+    private Boolean failOnError;
+    private String sqsUrl;
+    private Properties sqsClientProperties;
+    private SerializationSchema<InputT> serializationSchema;
+
+    SqsSinkBuilder() {}
+
+    /**
+     * Sets the url of the SQS that the sink will connect to. There is no default for this
+     * parameter, therefore, this must be provided at sink creation time otherwise the build will
+     * fail.
+     *
+     * @param sqsUrl the url of the Sqs
+     * @return {@link SqsSinkBuilder} itself
+     */
+    public SqsSinkBuilder<InputT> setSqsUrl(String sqsUrl) {
+        this.sqsUrl = sqsUrl;
+        return this;
+    }
+
+    /**
+     * Allows the user to specify a serialization schema to serialize each record to persist to SQS.
+     *
+     * @param schema serialization schema to use
+     * @return {@link SqsSinkBuilder} itself
+     */
+    public SqsSinkBuilder<InputT> setSerializationSchema(final SerializationSchema<InputT> schema) {
+        serializationSchema = schema;
+        return this;
+    }
```

Comment:
Should we instead provide a method to set the Element converter directly? Otherwise users will create SerializationSchema that takes type `InputT` -> `byte[]` -> `String` Rather than directly `InputT` -> `String`

Reply:
I tried to update the same, please let me know if that looks good

Reply:
Looks great - thanks @19priyadhingra !